src/Data/Time/PreciseDuration.purs

```diff
@@ -12,17 +12,47 @@ module Data.Time.PreciseDuration
   , toHours
   , toDays
   , toWeeks
+  , toDecimalLossy
+  , nano
+  , micro
+  , milli
+  , second
+  , minute
+  , hour
+  , day
+  , week
   ) where
 
 import Prelude
 
-import Data.BigInt (BigInt)
-import Data.BigInt as BigInt
 import Data.Decimal (Decimal)
 import Data.Decimal as Decimal
 
-import Data.Time.PreciseDuration.Internal (PreciseDuration(..), day, hour, micro, milli, minute, second, week)
-import Data.Time.PreciseDuration.Internal (PreciseDuration) as PreciseDurationType
+data PreciseDuration
+  = Nanoseconds Decimal
+  -- Nanoseconds must be integral, see smart constructor in Data.Time.PreciseDuration.
+  | Microseconds Decimal
+  | Milliseconds Decimal
+  | Seconds Decimal
+  | Minutes Decimal
+  | Hours Decimal
+  | Days Decimal
+  | Weeks Decimal
+
+derive instance eqPreciseDuration :: Eq PreciseDuration
+
+instance ordPreciseDuration :: Ord PreciseDuration where
+  compare x y = compare (toNanoseconds x) (toNanoseconds y)
```

Comment:
You could also do something like
```purescript
compare (Weeks a) (Weeks b) = compare a b
compare (Weeks a) b | Weeks b' <- toWeeks b = compare a b
...
```
Not sure if that will just be any more efficient though.